### ISSUE TITLE
Release Testing

### ISSUE THREAD

**Issue by jmchilton** [galaxyproject/planemo#184]
There have been repeated examples with releases which pass tests in repo but have resulted in released artifacts to PyPI to and Homebrew with serious problems.

 - Refactor the CLI integration tests to allow for optional testing against an external ``planemo`` command.
 - Extend ``make release`` with some real testing. After publishing to test PyPI:
   - Create a temp virtualenv.
   - Install all dependencies
   - Install planemo from test pypi.
   - run release tests.

**Comment by peterjc** [comment]
So this is the planemo self-tests which have not caught major problems?

**Comment by jmchilton** [comment]
Yeah - this is referring to planemo testing itself. The existing tox testing has missed relatively obvious problems such as 563963b7da26f8ef555aeb78dda1198916f4c4b6 ddd1643d76938c516b8965e031ceea040d9b3060 5963ed2e09aa17c856f5f21997e0b3ec1297386c (and that is just for the last couple weeks).
